rotkehlchen/accounting/pot.py

```diff
@@ -349,49 +362,70 @@ def get_prices_for_swap(
             return None
 
         try:
-            out_price = self.get_rate_in_profit_currency(
+            oracle_out_price = self.get_rate_in_profit_currency(
                 asset=asset_out,
                 timestamp=timestamp,
             )
         except (PriceQueryUnsupportedAsset, NoPriceForGivenTimestamp, RemoteError):
-            out_price = None
+            oracle_out_price = None
 
-        fee_price = None
-        if fee is not None and fee_asset is not None and fee != ZERO:
-            # also checking fee_asset != None due to https://github.com/rotki/rotki/issues/4172
+        oracle_fee_price = None
+        if fee_info is not None and fee_info.amount != ZERO:
             try:
-                fee_price = self.get_rate_in_profit_currency(
-                    asset=fee_asset,
+                oracle_fee_price = self.get_rate_in_profit_currency(
+                    asset=fee_info.asset,
                     timestamp=timestamp,
                 )
-            except (PriceQueryUnsupportedAsset, NoPriceForGivenTimestamp, RemoteError):
-                fee_price = None
+            except (PriceQueryUnsupportedAsset, RemoteError):
+                oracle_fee_price = None
+            except NoPriceForGivenTimestamp:
+                raise
+
         try:
-            in_price = self.get_rate_in_profit_currency(
+            oracle_in_price = self.get_rate_in_profit_currency(
                 asset=asset_in,
                 timestamp=timestamp,
             )
         except (PriceQueryUnsupportedAsset, RemoteError):
-            in_price = None
+            oracle_in_price = None
         except NoPriceForGivenTimestamp as e:
-            in_price = None
-            if e.rate_limited is True and out_price is None:
+            oracle_in_price = None
+            if e.rate_limited is True and oracle_out_price is None:
                 raise  # in_price = out_price = None -> notify user
 
-        if out_price is None and in_price is None:
+        if oracle_out_price is None and oracle_in_price is None:
             return None
 
-        if out_price is not None:
-            paid = amount_out * out_price
-            if fee_price is not None:
-                paid += fee_price * fee  # type: ignore # fee should exist here
-            calculated_in = Price(paid / amount_in)
+        # Calculate in price and optionally include a fee in this price
+        if oracle_out_price is not None:
+            total_paid = amount_out * oracle_out_price  # Use calculated_out_price
         else:
-            calculated_in = in_price  # type: ignore # in_price should exist here
+            total_paid = amount_in * oracle_in_price  # type: ignore  # in_price should exist here  # noqa: E501
```

Comment:
`oracle_in_price` can most definitely not exist here. The conditional on L396 only checks **if both are None**

Reply:
Yes. So when the execution comes to this block
```python
if oracle_out_price is not None:
    total_paid = amount_out * oracle_out_price
else:
    total_paid = amount_in * oracle_in_price 
```

we know for sure that at least one of `oracle_out_price` or `oracle_in_price` exists.

So if we come to this `else` block (line 403) we know that `oracle_out_price` is `None` and thus  `oracle_in_price` is NOT `None`.